python/test/t_GaussianNonLinearCalibration_std.py

```diff
@@ -54,3 +54,19 @@
     algo.setBootstrapSize(bootstrapSize)
     algo.run()
     print("result (Global)=", algo.getResult().getParameterMAP())
+
+# unobserved inputs
+p_ref = [2.8, 1.2, 0.5, 2.0]
+params = [0, 1, 2, 3]
+modelX = ot.ParametricFunction(model, params, p_ref)
+x = ot.Sample(m, 0)
```

Comment:
```suggestion
x = ot.Sample(m, 0)
y = modelX(x)
y += ot.Normal([0.0]*2, [0.05]*2, ot.IdentityMatrix(2)).getSample(m)
```
I believe that if you redefine the model (which you did by removing any input), then you also need to resample the observations, otherwise the observations are inconsistent with the model and calibration is simply impossible.

For the record, with my suggestion, you get the following result (proposed as a suggestion for `t_GaussianNonLinearCalibration_std.py` above): ``result (unobs.)= [1.64,2.5,0.644,0.871]``

Let us compare the model output with the true parameter `p_ref` to the model output with the calibrated parameters expected in the original `.expout` file, those produced by the original code and those produced after applying my suggestion.
```
>>> modelX.setParameter([2.8, 1.2, 0.5, 2.0]) # true parameter p_ref
>>> print(modelX([]))
[6.06,2.76]
>>> modelX.setParameter([2.63,1.75,1.09,2.75]) # original code output
>>> print(modelX([]))
[37.7,2.5]
>>> modelX.setParameter([2.8,2.18,1.5,1.85]) # original .expout
>>> print(modelX([]))
[37.8,2.39]
>>> modelX.setParameter([1.64,2.5,0.644,0.871]) # suggestion
>>> print(modelX([]))
[6.02,2.67]
```